code/Model/SiteTree.php

```diff
@@ -739,13 +739,16 @@ public function getAbsoluteLiveLink($includeStageEqualsLive = true)
     /**
      * Generates a link to edit this page in the CMS.
      *
+     * Implemented here to satisfy the CMSPreviewable interface, but data is intended to be loaded via Extension
+     *
+     * @see SilverStripe\Admin\CMSEditLinkExtension
+     *
      * @return string
      */
     public function CMSEditLink()
     {
-        // This method has to be implemented here to satisfy the CMSPreviewable interface.
-        // See the actual implementation in CMSEditLinkExtension.
-        return $this->extend('CMSEditLink')[0];
+        $extensionsResult = $this->extend('CMSEditLink');
+        return count($extensionsResult) ? $extensionsResult[0] : '';
```

Comment:
```suggestion
        return $extensionsResult[0] ?? '';
```
This is just a little clearer - as we don't really care about the number of items in the array so much as we care that there is a value in the `[0]` index.

Reply:
`extend` _always_ returns an `array`, never `null`.

Reply:
`??` squashes the error - basically says "If there's nothing there, use the next value"
It's called "null coalescing operator" but it actually does an `isset()` check.
https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op gives this as an explicitly recommended use case.

Reply:
Neat thanks, I knew it was null coalescing, but always forget if it also handles the "null pointer" reference too.